### PR TITLE
Danebook Authorization and Profile Editing

### PR DESCRIPTION
**Pull request by ayva**

### CODE REVIEW

File: app/helpers/profiles_helper.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module ProfilesHelper
+
+  def show_birthday(profile)
+    profile.user.birthday.nil? ? "No birthday specified" : profile.user.birthday.strftime("%B %d, %Y")
```
Comment (sicknarlo) [inline app/helpers/profiles_helper.rb:4]: I like the placeholder for when no data is present

File: spec/models/comment_spec.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
+describe Comment do 
+  let(:user){ build(:user) }
+  let(:comment){ build(:comment) }
+
+
+  # Happy
```
Comment (sicknarlo) [inline spec/models/comment_spec.rb:8]: Organizing them by happy/sad/bad is good practice, good job

File: spec/models/comment_spec.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  it "responds to author assosiation" do
+    expect(comment).to respond_to(:author)
+  end
+
+  it "responds to likes assosiation" do
+    expect(comment).to respond_to(:likes)
+  end
+
+  # Sad
+  it "deleted if author is deleted" do
```
Comment (sicknarlo) [inline spec/models/comment_spec.rb:23]: Does this work? If not, it may work if you run user.save! at some point. I don't think there is technically a user to destroy unless you do

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+describe User do
+  let(:user){ build(:user) }
+
+  # Happy
+  it "is valid with default attributes" do
+    expect(user).to be_valid
+  end
+
+  it "has password length bigger then 4" do
```
Comment (sicknarlo) [inline spec/models/user_spec.rb:11]: I think you may have these two mixed up; this test if the password is smaller than 4, but says bigger.

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    user.password = "1"
+    expect(user).to_not be_valid
+  end
+
+  it "has password length smaller then 24" do
```
Comment (sicknarlo) [inline spec/models/user_spec.rb:16]: Same with this one -- says testing smaller than 24, but actually tests larger.

File: spec/models/user_spec.rb (hunk 4 above)
Comment (sicknarlo) [inline spec/models/user_spec.rb:11]: I would test edge cases on these, meaning make sure 4 and 24 pass. That way you know it is actually testing if 4/24 are valid and just beyond (3/25) is not.

File: spec/models/user_spec.rb [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    user.password = "1"*25
+    expect(user).to_not be_valid    
+  end
+
+  it "has email with @ sign" do
```
Comment (sicknarlo) [inline spec/models/user_spec.rb:21]: Didn't think of this one as it was automatically done by one of the gems, good thinking! I'll have to add as well.
